frontend/src/theme/emotion.d.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @license
+ * Copyright 2018-2022 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Theme as StreamlitTheme } from "src/theme"
+
+// By default
+declare module "@emotion/react" {
+  export interface Theme extends StreamlitTheme {}
+}
```

Comment:
This may be me being a frontend dummy, but I don't understand what's happening here - do you mind adding an explanation comment?

Reply:
Yea. I can add a comment, but basically we are saying the "type of Theme" matches the type of our Streamlit theme type. This allows intellisense to get all of our values and typecheck against our custom theme structure and not an `any` or a hyperspecific defined theme they might have.

See https://emotion.sh/docs/typescript#define-a-theme